src/libkrun/src/lib.rs

```diff
@@ -156,23 +158,36 @@ impl ContextConfig {
     }
 
     #[cfg(feature = "blk")]
-    fn set_root_block_cfg(&mut self, block_cfg: BlockDeviceConfig) {
-        self.root_block_cfg = Some(block_cfg);
+    fn add_block_cfg(&mut self, block_cfg: BlockDeviceConfig) {
+        self.block_cfgs.push(block_cfg);
     }
 
     #[cfg(feature = "blk")]
-    fn get_root_block_cfg(&self) -> Option<BlockDeviceConfig> {
-        self.root_block_cfg.clone()
+    fn set_root_block_cfg(&mut self, block_cfg: BlockDeviceConfig) {
+        self.root_block_cfg = block_cfg;
     }
 
     #[cfg(feature = "blk")]
     fn set_data_block_cfg(&mut self, block_cfg: BlockDeviceConfig) {
-        self.data_block_cfg = Some(block_cfg);
+        self.data_block_cfg = block_cfg;
     }
 
     #[cfg(feature = "blk")]
-    fn get_data_block_cfg(&self) -> Option<BlockDeviceConfig> {
-        self.data_block_cfg.clone()
+    fn get_block_cfg(&self) -> Vec<BlockDeviceConfig> {
+        // Blocks added with the new API
+        let cfgs = self.block_cfgs.clone();
+
+        // Blocks added with the old API. For backwards compat, when cfgs is empty (the new API is
+        // not used), this needs to be root and then data, in that order. Also for backwards
+        // compat, root/data are setters and need to discard redundant calls. So we have simple
+        // setters above and fix up here.
+        for cfg in [self.root_block_cfg, self.data_block_cfg] {
+            if let Some(cfg) = cfg {
+                cfgs.push(cfg);
+            }
+        }
+
+        cfgs
```

Comment:
To conform with the comment, I'd suggest replacing this with something like:

```
        if self.block_cfgs.is_empty() {
            [&self.root_block_cfg, &self.data_block_cfg]
                .into_iter()
                .filter_map(|cfg| cfg.clone())
                .collect()
        } else {
            self.block_cfgs.clone()
        }
```

Reply:
done.